openeo/local/connection.py

```diff
@@ -0,0 +1,290 @@
+import datetime
+import xarray as xr
+import numpy as np
+import rioxarray
+from pathlib import Path
+from pyproj import Transformer
+from typing import Dict, List, Tuple, Union, Callable, Optional, Any, Iterator
+
+from openeo.metadata import CollectionMetadata
+from openeo.internal.graph_building import PGNode
+from openeo.rest.datacube import DataCube
+from openeo.internal.jupyter import VisualDict, VisualList
+
+def _get_dimension(dims: dict, candidates: List[str]):
+    for name in candidates:
+        if name in dims:
+            return name
+        
+def _get_netcdf_zarr_metadata(file_path):
```

Comment:
all this netcdf and geotiff handling should probably move outside of "connection.py" module because it is not directly related to the connection object.
move it e.g. to "collections.py"?